## Reviewer verdict: REQUEST CHANGES

### Blocker — lint failure on line 29
\`cmd/gc/gc_beads_bd_lint_test.go:29:15: Error return value of \`f.Close\` is not checked (errcheck)\`

\`defer f.Close()\` doesn't comply with this repo's errcheck config. Project convention (see [`cmd/gc/bd_env_test.go:259`](https://github.com/gastownhall/gascity/blob/main/cmd/gc/bd_env_test.go#L259)):
\`\`\`go
defer func() { _ = f.Close() }() //nolint:errcheck // test cleanup
\`\`\`

This single line failure cascades into `Preflight / lint`, `CI / preflight`, `CI / required`, `Check`, and `CI / integration` (5 of the 6 red checks). Fixing it should clear all 5.

### Non-blocker — flake
`Integration / rest-full-8-of-8` failed `TestGCLiveContract_BeadsAndEvents` with `context deadline exceeded` on a SSE stream — same flake seen on PR #1609. Unrelated to this diff (this PR is a pure-fs string-matcher test). Will be re-evaluated on the next push.

### Otherwise
Test logic is sound, comment explains the *why* (ga-5mym, `ensure_bd_runtime_config_value`), regex covers both `bd <args> config set` and `run_bd_<wrapper>` shapes.

--- a/cmd/gc/gc_beads_bd_lint_test.go
+++ b/cmd/gc/gc_beads_bd_lint_test.go
@@ -26,7 +26,7 @@ func TestGcBeadsBdNoBdConfigSet(t *testing.T) {
 	if err != nil {
 		t.Fatalf("open script: %v", err)
 	}
-	defer f.Close()
+	defer func() { _ = f.Close() }() //nolint:errcheck // test cleanup
 
 	// Match `bd <args> config set` and `run_bd_<wrapper> <args> config set`.
 	// The script invokes bd both directly and through helpers like

